src/StateMachine.jl

```diff
@@ -240,5 +221,105 @@ function pretty(
         end
         long && print(io, "   ")
         print(io, " ELSE: $(name(next[2], id))")
+    elseif next isa Tuple{}
+        if long
+            print(io, "\n   ")
+            print(io, " END")
+        end
+    else
+        error(" UNKNOWN ")
+    end
+end
+
+# We merge states with identical behavior, bottom-up, to minimize the size of
+# the state machine.  We define `hash` and `==` to take account of only what matters.
+# Specifically, we ignore the `cases::ImmutableVector{CasePartialResult}` of `CodePoint`.
+mutable struct DeduplicatedCodePoint
```

Comment:
would it make sense to create an `AbstractCodePoint` to be the parent of this and code point, then that can be used in some of the methods above to bind the `T <: AbstractCodePoint`?

Reply:
Sure.